src/Symfony/Bundle/FeatureToggleBundle/DataCollector/FeatureCheckerDataCollector.php

```diff
@@ -0,0 +1,156 @@
+ * } $data
+ */
+final class FeatureCheckerDataCollector extends DataCollector implements LateDataCollectorInterface
+{
+    /** @var \SplStack<string> */
+    private \SplStack $currentToggle;
+
+    /** @var \SplStack<string> */
+    private \SplStack $currentCompute;
+
+    public function __construct(
+        private readonly FeatureCollection $featureCollection,
+    ) {
+        $this->data = ['features' => [], 'toggles' => []];
+        $this->currentToggle = new \SplStack();
+        $this->currentCompute = new \SplStack();
+    }
+
+    public function collect(Request $request, Response $response, \Throwable $exception = null): void
+    {
+        foreach ($this->featureCollection as $feature) {
+            $strategy = (\Closure::bind(fn(): StrategyInterface => $this->strategy, $feature, Feature::class))();
+            $default = (\Closure::bind(fn(): bool => $this->default, $feature, Feature::class))();
+
+            $this->data['features'][$feature->getName()] = [
+                'default' => $default,
+                'description' => $feature->getDescription(),
+                'strategy' => $strategy,
+            ];
+        }
+    }
+
+    public function collectIsEnabledStart(string $featureName): void
+    {
+        $toggleId = uniqid();
+
+        $this->data['toggles'][$toggleId] = [
+            'feature' => $featureName,
+            'computes' => [],
+            'result' => null,
+        ];
+        $this->currentToggle->push($toggleId);
+    }
+
+    /**
+     * @param class-string $strategyClass
+     */
+    public function collectComputeStart(string $strategyId, string $strategyClass): void
+    {
+        $toggleId = $this->currentToggle->top();
+        $computeId = uniqid();
+        $level = $this->currentCompute->count();
+
+        $this->data['toggles'][$toggleId]['computes'][$computeId] = [
+            'strategyId' => $strategyId,
+            'strategyClass' => new ClassStub($strategyClass),
+            'level' => $level,
+            'result' => null,
+        ];
+        $this->currentCompute->push($computeId);
+    }
+
+    public function collectComputeStop(StrategyResult $result): void
+    {
+        $toggleId = $this->currentToggle->top();
+        $computeId = $this->currentCompute->pop();
+
+        $this->data['toggles'][$toggleId]['computes'][$computeId]['result'] = $result;
+    }
+
+    public function collectIsEnabledStop(bool $result): void
+    {
+        $toggleId = $this->currentToggle->pop();
+
+        $this->data['toggles'][$toggleId]['result'] = $result;
+    }
+
+    public function getName(): string
+    {
+        return 'feature_toggle';
+    }
+
+    public function reset(): void
+    {
+        $this->data = [
+            'features' => [],
+            'toggles' => [],
+        ];
+    }
+
+    public function lateCollect(): void
+    {
+        $this->data = $this->cloneVar($this->data);
+    }
+
+    /**
+     * @return list<FeatureType>|Data
```

Comment:
using a non-prefixed `@return` using a phpstan type alias is a no-go as other tools would interpret that as returning a list of objects of a non-existent class FeatureType.

and in Symfony, we don't use phpstan type aliases.

Reply:
Got it. So it should be `@phpstan-return` ? Cannot seem to find the documentation for that one.

Reply:
When using a phpstan type alias, the phpdoc tags should indeed use the `phpstan-` prefix. This way, a tool than does not know about phpstan would not read them at all (a tool reading `phpstan-return` without supporting the `phpstan-type` would be a buggy tool, and it would be their fault).

However, in this case, this is not the solution because of the second paragraph in my previous message.

Reply:
fixed